### PR TITLE
added profile and home page,UserController

### PR DESCRIPTION
**Pull request by xerxes-on**

### CODE REVIEW

File: app/Http/Controllers/Api/BookController.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+use App\Models\Book;
+use Illuminate\Http\Request;
+
+class BookController
+{
+    public function index()
+    {
+         $books = Book::with('author')->paginate(10);
+         $count = Book::count();
+         return response(BooksResource::collection($books), $count);
```
Comment (YauheniNikifarau) [inline app/Http/Controllers/Api/BookController.php:15]: Second argument of response() helper should be response code

File: app/Http/Controllers/Api/QuoteController.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+     * Store a newly created resource in storage.
+     */
+    public function store(Request $request)
+    {
+        $request->validate([
+            'quote'=> 'required|min:10|unique:'. Quote::class,
+            'author_id'=>'required',
+        ]);
+        $result = Quote::create($request->all());
+        return response($result, 401);
```
Comment (YauheniNikifarau) [inline app/Http/Controllers/Api/QuoteController.php:32]: why 401?

File: app/Http/Controllers/Api/QuoteController.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    /**
+     * Store a newly created resource in storage.
+     */
+    public function store(Request $request)
+    {
+        $request->validate([
+            'quote'=> 'required|min:10|unique:'. Quote::class,
+            'author_id'=>'required',
+        ]);
+        $result = Quote::create($request->all());
```
Comment (YauheniNikifarau) [inline app/Http/Controllers/Api/QuoteController.php:31]: passing $request->all() to create method is not the best practise. There is DTO recommended to be used.

File: app/Http/Controllers/Api/QuoteController.php [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    /**
+     * Display the specified resource.
+     */
+    public function show(string $id)
+    {
+        $quote = Quote::with('users', 'author')->find($id);
+        if ($quote) {
+            return response(new QuotesResource($quote));
+        } else {
+            return ('There is no hotel with such id');
```
Comment (YauheniNikifarau) [inline app/Http/Controllers/Api/QuoteController.php:44]: Seems wrong. 'response' missing

File: app/Http/Controllers/Api/QuoteController.php [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+     * Update the specified resource in storage.
+     */
+    public function update(Request $request, $id)
+    {
+        $quote = Quote::find($id);
+        $request->validate([
+            'quote'=> 'required|min:10|unique:'. Quote::class,
+            'author_id'=>'required',
+        ]);
+        $quote->update($request->all());
```
Comment (YauheniNikifarau) [inline app/Http/Controllers/Api/QuoteController.php:58]: See below about $response->all()

File: app/Http/Controllers/Api/QuoteController.php [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+    /**
+     * Remove the specified resource from storage.
+     */
+    public function destroy(string $id)
+    {
+        $quote = Quote::find($id);
+        if($quote){
+             $quote->delete();
+            return response('deleted', 200);
```
Comment (YauheniNikifarau) [inline app/Http/Controllers/Api/QuoteController.php:70]: code 200 is less suitable here, other code is more suitable

File: app/Models/Rating.php [hunk 7]
```diff
@@ -8,12 +8,16 @@
...
 {
     use HasFactory;
+    protected $fillable = [
+        'rating', 'user_id', 'comment'
+    ];
     public function book()
     {
         return $this->belongsTo(Book::class);
     }
     public function user()
```
Comment (YauheniNikifarau) [inline app/Models/Rating.php:18]: Bette to add Response type here

**Review by YauheniNikifarau — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/Api/QuoteController.php (hunk 2 above)
Comment (xerxes-on) [reply app/Http/Controllers/Api/QuoteController.php:32]: it should be 200( success) my mistake

File: app/Models/Rating.php (hunk 7 above)
Comment (xerxes-on) [reply app/Models/Rating.php:18]: what about $guarded = []; ?

File: app/Http/Controllers/Api/QuoteController.php (hunk 6 above)
Comment (xerxes-on) [reply app/Http/Controllers/Api/QuoteController.php:70]: Ok

**Pull request merged**